Tests/SentryTests/Networking/SentryHTTPInterceptorTests.swift

```diff
@@ -0,0 +1,256 @@
+import XCTest
+
+class SentryHTTPInterceptorTests: XCTestCase {
+    
+    private static let httpUrl = "http://somedomain.com"
+    private static let httpsUrl = "https://somedomain.com"
+    private static let wsUrl = "ws://somedomain.com"
+    private static let dsnAsString = TestConstants.dsnAsString(username: "SentrySessionTrackerTests")
+    
+    class TestURLSession: URLSession {
+        var invalidateAndCancelDate: Date?
+        var lastDataTask: URLSessionDataTaskMock?
+        
+        override func dataTask(with request: URLRequest) -> URLSessionDataTask {
+            lastDataTask = URLSessionDataTaskMock(request: request)
+            return lastDataTask!
+        }
+        
+        override func invalidateAndCancel() {
+            invalidateAndCancelDate = CurrentDate.date()
+        }
+    }
+    
+    class TestSentryHTTPInterceptor: SentryHttpInterceptor {
+        override func createSession() -> URLSession {
+            return TestURLSession()
+        }
+    }
+    
+    private class Fixture {
+        let client: TestClient
+        let hub: TestHub
+        let scope: Scope
+        let span: Span
+        let request: URLRequest!
+        let options: Options
+        let dateProvider = TestCurrentDateProvider()
+        
+        init() {
+            options = Options()
+            options.dsn = SentryHTTPInterceptorTests.dsnAsString
+            
+            scope = Scope()
+            client = TestClient(options: options)!
+            span = SentrySpan(context: SpanContext(operation: "SomeOperation"))
+            hub = TestHub(client: client, andScope: scope)
+            scope.span = span
+            request = URLRequest(url: URL(string: SentryHTTPInterceptorTests.httpUrl)!)
+            CurrentDate.setCurrentDateProvider(dateProvider)
+        }
+        
+        func getSut() -> SentryHttpInterceptor {
+            return TestSentryHTTPInterceptor(request: request, cachedResponse: nil, client: TestProtocolClient())
+        }
+    }
+    
+    private class AlternativeTestProtocol: URLProtocol {
+        
+    }
+    
+    private var fixture: Fixture!
+    
+    override func setUp() {
+        super.setUp()
+        
+        fixture = Fixture()
+        SentrySDK.setCurrentHub(fixture.hub)
+    }
+    
+    override func tearDown() {
+        super.tearDown()
+        SentrySDK.setCurrentHub(nil)
```

Comment:
`m`: We have a method now clearing everything.
```suggestion
        clearTestState()
```